platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/XMLUtils.java

```diff
@@ -124,28 +135,113 @@ public static Result transform(Node nodeXml, TransformerProperties transformerPr
+        try {
+            TransformerFactory transFactory = TransformerFactory.newInstance();
+            Transformer transformer = transFactory.newTransformer();
+
+            for (Entry<String, String> entry : transformProperties.getOutputProperties()) {
+                transformer.setOutputProperty(entry.getKey(), entry.getValue());
+            }
+            if (transformProperties.getErrorListener() != null) {
+                transformer.setErrorListener(transformProperties.getErrorListener());
+            }
+            transformer.transform(sourceXml, result);
+        } catch (TransformerException e) {
+            LOGGER.debug("Unable to transform XML.", e);
+        } finally {
+            Thread.currentThread()
+                    .setContextClassLoader(tccl);
+        }
+    }
+
+    /**
+     * Iterate through the elements of an XML document.
+     * <p>
+     * It uses a lambda function and an instance of the the internal class ResultHolder. The result
+     * stores the return value. The lambda function has a change to process every element
+     * in the document. To store a result, call set() on the result object.
+     * <p>
+     * If the lambda function returns false, processing continues to the next element.
+     * <p>
+     * When the last element in the document is processed, the value in the holder object
+     * is returned.
+     * <p>
+     * If the function encounters a processing exception and the value returned to the caller is null.
+     * <p>
+     * If the lambda function return true, processing terminate processing, During processing Set the value in the ProcessingResult to be the return value of this method.
+     * To stop processing, set the value "done" in ProcessingResult to true.
+     * For an example, see the getRootNamespace() method.
+     *
+     * @param xml                    The XML to process
+     * @param processElementFunction Function that accepts an instance of XMLStreamReader and result holder. The function must return a boolean.
+     * @return <T>  The result of the processing
+     */
 
+    public static <T> T processElements(String xml,
+            BiFunction<ResultHolder<T>, XMLStreamReader, Boolean> processElementFunction) {
+
+        initializeXMLInputFactory();
         XMLStreamReader xmlStreamReader = null;
+        ResultHolder<T> result = new ResultHolder<>();
+        boolean keepProcessing;
 
         try (StringReader strReader = new StringReader(xml)) {
             xmlStreamReader = xmlInputFactory.createXMLStreamReader(strReader);
-
             while (xmlStreamReader.hasNext()) {
                 int event = xmlStreamReader.next();
                 if (event == XMLStreamConstants.START_ELEMENT) {
-                    rootNamespace = xmlStreamReader.getNamespaceURI();
-                    return rootNamespace;
+                    keepProcessing = processElementFunction.apply(result, xmlStreamReader);
+                    if (!keepProcessing) {
```

Comment:
As this is the last thing in this loop, could probably change to initialize `keepProcessing` to true and then...
```java
while (keepProcessing && xmlStreamReader.hasNext()) {
...
```
and skip the test and break inside the loop.